lib/auth/webauthn/httpserver/main.go

```diff
@@ -0,0 +1,321 @@
+// Copyright 2021 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
```

Comment:
Is it weird to have a `package main` under the `lib/` directory?

Reply:
I'm the one who put it here, so I'm going to say no. :)

There are a few reasons why I think it's ok:

1) It is meant for maintainers of the webauthn impl, not end-users, so it's tucked away in a somewhat hidden place (but easy for someone working on webauthn to find)

2) It shares fate with the webauthn package - if you delete/move/etc the former, this one goes along.

Placing under ./tools is a big no for me, ./examples doesn't seem right either. Looking around a bit more I found ./lib/srv/desktop/rdp/rdpclient/testclient/main.go, which seems to take a similar path to this one.

Reply:
I am not sure if it makes sense to check in this sample server under `/lib`.  Once the changes are addressed on the client side (web app), this code is no longer needed. In general I am against adding any code used for local development this way.

Reply:
UI integration does help, but it hasn't started yet. This is very useful to quickly get real-world responses from browsers (and throwing them at Teleport). It is also much simpler to tweak than the UI would be.

I can keep it for myself, but it sounds better to have it in. I do agree that if it becomes redundant or obsolete that we should delete it.

What do you say, @alex-kovoy? Could we merge this, at least for now?

Reply:
the same applies for `https://github.com/gravitational/teleport/tree/master/lib/srv/desktop/rdp/rdpclient/testclient` as currently it's been implemented for the frontend.

Reply:
Ack.

Could we get this in, though? It makes life much easier and I'd rather capture in the repo, at least during this stage.